### PR TITLE
MODINVSTOR-710: Suspend reindex stream when producer write queue is full.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
## Changes:

* Add `folio-kafka-wrapper` dependency.
* Remove unnecessary code that is now used from the `folio-kafka-wrapper`.
* Use separate shared producer per topic and separate producer when stream published.
* Pause reindex stream when producer write queue is full.
* Simplify publishing approach.

### CODE REVIEW

**Review by pfilippovEPAM — APPROVED** [review]

File: src/main/java/org/folio/services/kafka/KafkaProperties.java [hunk 1]
```diff
@@ -1,44 +1,29 @@
...
-import static java.lang.Integer.parseInt;
 import static java.lang.Short.parseShort;
 import static java.lang.System.getenv;
 
-import java.util.Map;
-import org.apache.kafka.common.serialization.StringSerializer;
-
 public final class KafkaProperties {
-  private static int port = parseInt(getenv().getOrDefault("KAFKA_PORT", "9092"));
+  private static String port = getenv().getOrDefault("KAFKA_PORT", "9092");
```
Comment (hjiebsco) [inline src/main/java/org/folio/services/kafka/KafkaProperties.java:7]: Any advantage to use `string` over `int` for `port`, or just try to match with `KafkaConfig` in `folio-kafka-wrapper`?

File: src/main/java/org/folio/services/reindex/ReindexJobRunner.java [hunk 2]
```diff
@@ -1,53 +1,59 @@
...
+      new CommonDomainEventPublisher<>(vertxContext, okapiHeaders, INVENTORY_INSTANCE),
+      tenantId(okapiHeaders));
   }
 
-  public ReindexJobRunner(InstanceDomainEventPublisher publisher,
-    PostgresClientFuturized postgresClient, ReindexJobRepository repository,
-    Context vertxContext) {
+  public ReindexJobRunner(PostgresClientFuturized postgresClient, ReindexJobRepository repository,
+    Context vertxContext, CommonDomainEventPublisher<Instance> domainEventPublisher,
+    String tenantId) {
```
Comment (hjiebsco) [inline src/main/java/org/folio/services/reindex/ReindexJobRunner.java:51]: Is `public` just for test convenience?

**Review by hjiebsco — APPROVED** [review]

File: src/main/java/org/folio/services/kafka/KafkaProperties.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/services/kafka/KafkaProperties.java:7]: I tried to match the `folio-kafka-wrapper`

File: src/main/java/org/folio/services/reindex/ReindexJobRunner.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/services/reindex/ReindexJobRunner.java:51]: Yes, the constructor is for tests.

**Pull request merged**